hanadb_exporter/db_manager.py

```diff
@@ -139,3 +145,42 @@ def get_connectors(self):
         Get the connectors
         """
         return self._db_connectors
+
+    def get_active_master_host(self, hosts, master_host, master_port):
+        """
+        Check if the connection to MASTER host is up, otherwise find and return the current active MASTER host.
+        """
+        if self._system_db_connector.isconnected():
+            return master_host
+        else:
+            self._logger.info('Master host=%s is not active, scale-out is enable, looking for new Master..', str(master_host))
+            current_time = time.time()
+            timeout = current_time + SCALE_OUT_TIME_OUT
+            while current_time <= timeout:
+                for host in hosts:
+                    conn = hdb_connector.HdbConnector()
+                    try:
+                        conn.connect(host, master_port, self._connection_data)
+                        self._logger.info('System database is up on a new host: %s', str(host))
+                        return host
+                    except hdb_connector.connectors.base_connector.ConnectionError:
+                        pass
+
+                time.sleep(RECONNECTION_INTERVAL)
+                current_time = time.time()
+            else:
+                raise hdb_connector.connectors.base_connector.ConnectionError(
+                    'Unable to connect to system database on any host.')
+
+    def get_database_hosts(self):
+        """
+        Get database hosts
+        """
+        hosts = []
+        data = self._system_db_connector.query(self.HOSTS_DATA_QUERY)
+        formatted_data = utils.format_query_result(data)
+        for row_data in formatted_data:
+            hosts.append(row_data['HOST'])
```

Comment:
Maybe we should have some exception handling in this `HOST` lookup, if for some reason the row comes empty

Reply:
ok I will add that